hugo/layouts/partials/footer.html

```
@@ -79,17 +79,5 @@ <h2 class="scilife-footer-titles mt-2 mb-1">Stay Updated</h2>
 </footer>
 
 <!-- Matomo tracking info -->
-<script>
-    var _paq = window._paq = window._paq || [];
-    /* tracker methods like "setCustomDimension" should be called before "trackPageView" */
-    _paq.push(["disableCookies"]);
-    _paq.push(['trackPageView']);
-    _paq.push(['enableLinkTracking']);
-    (function () {
-        var u = "//matomo.dc.scilifelab.se/";
-        _paq.push(['setTrackerUrl', u + 'matomo.php']);
-        _paq.push(['setSiteId', '11']);
-        var d = document, g = d.createElement('script'), s = d.getElementsByTagName('script')[0];
-        g.async = true; g.src = u + 'matomo.js'; s.parentNode.insertBefore(g, s);
-    })();
-</script>
+{{ $matomoJS := resources.Get "js/matomo.js" | fingerprint | minify }}
```

A Hugo idiom I like for all these cases:
```
{{ with resources.Get "js/matomo.js" | fingerprint | minify }}
  <script src="{{ .RelPermalink }}"></script>
{{ else }}
  {/* Error handling */}
{{ end }}
```

---

If the file doesn't exist the build will fail because of the fingerprint step, so the above `{{ else }}` block would never be triggered. We could do something like this I suppose:

```
{{ $jsFile := "js/does_not_exist.js" }}
{{ $resource := resources.Get $jsFile }}
{{ if $resource }}
  {{ $minified := $resource | fingerprint | minify }}
  <script src="{{ $minified.RelPermalink }}"></script>
{{ else }}
  <script>
    console.error("Failed to load Matomo tracking script: {{ $jsFile }}.");
  </script>
{{ end }}
```
But perhaps it is better to let the build fail and fix the file path?

---

I see your point, in this case I suppose I lean more towards not adding the custom error handling because I think the default error you would get is sufficient to explain the problem. But it doesn't matter now as your other suggestion meant that all this code is no longer needed haha.